sphinx_gallery/tests/test_full.py

```diff
@@ -241,14 +241,17 @@ def test_image_formats(sphinx_app):
             assert op.isfile(file_fname), file_fname
             want_html = 'src="%s"' % (img_fname0,)
             assert want_html in html
+            img_fname2 = ('../_images/sphx_glr_%s_%03d_2_0x.%s' %
+                          (ex, num, ext))
+            file_fname2 = op.join(generated_examples_dir, img_fname2)
+            want_html = 'srcset="%s, %s 2.0x"' % (img_fname0, img_fname2)
             if ext in ('png', 'jpg'):  # check 2.0x file.  tests the directive
-                img_fname2 = ('../_images/sphx_glr_%s_%03d_2_0x.%s' %
-                              (ex, num, ext))
-                file_fname2 = op.join(generated_examples_dir, img_fname2)
                 assert op.isfile(file_fname2), file_fname2
-
-                want_html = 'srcset="%s, %s 2.0x"' % (img_fname0, img_fname2)
                 assert want_html in html
+            else:
+                assert ext == 'svg'
+                assert not op.isfile(file_fname2), file_fname2
```

Comment:
... ha, this test caught a bug.  This file *was* being made...

Reply:
Well, hmm, I guess this is a bit of a question - dpi does mean something for rasterized artists in an svg.  So what I was doing was making a 100-dpi svg and a 200-dpi svg.  Of course if there were no rasterized artists they are the same.  But if there are rasterized artists, they can be different.  

So, is it worth making svg behave differently?

Reply:
I don't know enough about HTML/CSS to know for sure, but based on some quick reading just now it seems like SVG shouldn't need multiple versions. I would try with and without multiple versions and use different devices in Chrome's developer mode (F12) where you can set the pixel device ratio explicitly. In theory some of the builtin ones should have different ratios, but it looks like you can set it explicitly in a custom device so I'd go that route:

![Screenshot from 2021-04-21 10-32-18](https://user-images.githubusercontent.com/2365790/115571351-e79ed780-a28c-11eb-88ee-83341cd0e0b3.png)

Reply:
Well I tried using different pixel ratios there and it didn't make any difference when I toggled

Reply:
Pngs change though?   

The svg in tiny doesn't have any rasterized artists so I wouldn't expect a difference in the svg.  But it is possible to create svg with rasterized artists like pcolormesh or image.

Reply:
Actually it looks like you can do it just by making the top-bar have the ratio option, and refreshing -- it correctly pulls the 2x source but looks identical to me (not surprising I guess):

![Screenshot from 2021-04-21 10-38-00](https://user-images.githubusercontent.com/2365790/115572266-bd014e80-a28d-11eb-965c-a01ecb20204a.png)

At least this one SVG one looks identical as well:

![Screenshot from 2021-04-21 10-40-11](https://user-images.githubusercontent.com/2365790/115572540-fe91f980-a28d-11eb-8ac9-07b118d2fac0.png)

Reply:
The PNGs do seem to change (even though I can't detect it by eye), at least acconding to the `currentSrc` that shows up.

> The svg in tiny doesn't have any rasterized artists so I wouldn't expect a difference in the svg. But it is possible to create svg with rasterized artists like pcolormesh or image.

Ahh that makes sense, hadn't considered the rasterized parts. Then I suppose it does make sense to have the additional sizes

Reply:
On my retina display there is a pretty clear difference in the websites before and after ;-)  Though I will admit the default matplotlib line widths end up looking a little spindly!